src/components/Tutor/BaseURL.js

```diff
@@ -0,0 +1,68 @@
+function isValid(val) {
+  return val !== "";
+}
+
+export class BaseURL {
+  URL = `${process.env.REACT_APP_MERAKI_URL}/volunteers`;
+
+  /*Setting date range */
+  setDates(fromStart, toEnd) {
+    this.URL = this.URL.concat(`?from=${fromStart}&to=${toEnd}`);
+  }
+
+  /*Filters and Search */
+  /*  Total varialbles = n = 3
+        Number of cases = nC3 + nC2 + nC1 = 1 + 3 + 3 = 7
+    */
+  setFilterSearch(statusFilter, searchTerm, langFilter) {
+    switch (true) {
+      /*All 3 are present*/
+      case isValid(statusFilter) && isValid(searchTerm) && isValid(langFilter):
+        this.URL = this.URL.concat(
+          `?status=${statusFilter}&name=${searchTerm}&lang=${langFilter}`
+        );
+        break;
+      /*Only status and search present*/
+      case isValid(statusFilter) && isValid(searchTerm):
+        this.URL = this.URL.concat(
+          `?status=${statusFilter}&name=${searchTerm}`
+        );
+        break;
+      /*Only search and language present*/
+      case isValid(searchTerm) && isValid(langFilter):
+        this.URL = this.URL.concat(`?name=${searchTerm}&lang=${langFilter}`);
+        break;
+      /*Only status and language present*/
+      case isValid(statusFilter) && isValid(langFilter):
+        this.URL = this.URL.concat(
+          `?status=${statusFilter}&lang=${langFilter}`
+        );
+        break;
+      /*Only status present*/
+      case isValid(statusFilter):
+        this.URL = this.URL.concat(`?status=${statusFilter}`);
+        break;
+      /*Only search present*/
+      case isValid(searchTerm):
+        this.URL = this.URL.concat(`?name=${searchTerm}`);
+        break;
+      /*Only language present*/
+      case isValid(langFilter):
+        this.URL = this.URL.concat(`?lang=${langFilter}`);
+        break;
+    }
+  }
```

Comment:
What if you want to add more filters in the future? Every time you add 1, you're going to be doubling the number of possibilities. This will mean more inputs to `statusFilter` and exponentially growing duplication of code. It'd be better to use an object input with keys and values (e.g., `{status: statusFilter, lang: langFilter, name: searchTerm}`). But for now, you could still remove the code duplication:
```suggestion
    const filters = {status: statusFilter, lang: langFilter, name: searchTerm};
    this.URL += "?" + Object.entries(filters)
      .filter((keyValuePair) => isValid(keyValuePair[1]))
      .map((keyValuePair) => `${keyValuePair[0]}=${keyValuePair[1]}`)
      .join("&");
```
Another possibility is to use `toString` on a `URLSearchParams` object, but you'd still need to `filter` out the invalid properties first.